DirectEmployers/MyJobs
# [PD-3025] Group ReportingTypes on report intentions page.

## What's this PR do?
Allows for grouping ReportingTypes so that the user can understand what they're looking at and find the reporting type they're looking for more easily.

## Screenshots (if appropriate)
![report_intentions](https://cloud.githubusercontent.com/assets/2728248/24155774/1ff028dc-0e2b-11e7-9e83-3e8b0d2530c6.png)

## Where should the reviewer start?
* The view in `myreports/views.py` has been updated to return a nested dictionary of groups and reporting types, where previously it was only returning the reporting types. The outer level is the group, and the inner level contains group information, and a list of matching report types.
* The javascript has been updated so that, if a group is present, the reporting types are grouped together. Otherwise the reporting types should appear to be their own separate entity (as it was before).
* New reporting types, from start to finish, should be manageable in the Django admin. Unfortunately, due to limitations of the django admin, there's quite a few separate pages required to make this happen.
** I didn't add history to the objects like I jokingly said I might. Yet.

## How should this be manually tested?

1. Run both migrations
2. Navigate to https://secure.my.jobs/reports/view/overview/ and check out the awesome new reporting intentions page.
3. Go to the Django admin and create your own new reports.
4. Navigate back to the page in step (2) and see how much more awesome the reporting intentions page got.

## Any background context you want to provide?
No.
